app.js

```diff
@@ -94,11 +106,35 @@ app.on('ready', () => {
         }
     });
 
+    async function setActivity() {
+        if (!rpc || !mainWindow) {
+            return;
+        }
+
+        rpc.setActivity({
+            startTimestamp,
+            largeImageKey: 'rift',
+            largeImageText: 'Rift Explorer',
+            instance: false,
+        });
+    }
+
+    rpc.on('ready', () => {
+        setActivity();
+
+        // activity can only be set every 15 seconds
+        setInterval(() => {
+            setActivity();
+        }, 15e3);
+    });
+
     connector.start();
-});
+    rpc.login({ clientId }).catch(console.error);
+ });
 
 app.on('window-all-closed', () => {
     if (process.platform !== 'darwin') {
+        rpc.destroy();
```

Comment:
Why is this under the if? Does it not need to be destroyed under macOS?

Reply:
My bad human error.